pylivetrader/backend/alpaca.py

```diff
@@ -549,50 +556,106 @@ def _symbol_bars(
+
+        if not (_from and to):
+            _from, to = self._get_from_and_to(size, limit, end_dt=to)
+
+        if self._use_polygon:
+            args = [[symbol, _from, to, size]for symbol in symbols]
+            return parallelize_with_multi_process(self._fetch)(args)
+        else:
+            # alpaca support get real-time data of multi stocks(<200) at once
+            splitlen = 199
+            parts = []
+            for i in range(0, len(symbols), splitlen):
+                part = symbols[i:i + splitlen]
+                parts.append(part)
+            args = [[part, _from, to, size] for part in parts]
+            return parallelize_with_multi_process(self._fetch, 10)(args)
+            # return parallelize(self._fetch)(args).values()
+
+    def _symbol_trades(self, symbols):
+        '''
+        Query last_trade in parallel for multiple symbols and
+        return in dict.
+
+        symbols: list[str]
+
+        return: dict[str -> polygon.Trade]
+        '''
 
         @skip_http_error((404, 504))
         def fetch(symbol):
             if self._use_polygon:
+                return self._api.polygon.last_trade(symbol)
+            else:
+                return self._api.get_last_trade(symbol)
+
+        return parallelize(fetch)(symbols)
+
+    def _get_from_and_to(self, size, limit, end_dt=None):
+        """
+        generate time interval for api
+
+        return:pd.Timestamp(tz=America/New_York)
+        """
+        if not end_dt:
+            end_dt = pd.to_datetime('now', utc=True).floor('min')
+        session_label = self._cal.minute_to_session_label(end_dt)
+        all_minutes: pd.DatetimeIndex = self._cal.all_minutes
+        all_sessions: pd.DatetimeIndex = self._cal.all_sessions
+        if size == 'minute':
+            if end_dt not in self._cal.minutes_for_session(session_label):
+                end_dt = self._cal.previous_minute(end_dt)
+            idx = all_minutes.get_loc(end_dt)
+            start_minute = all_minutes[idx - limit + 1]
+            _from = end_dt.tz_convert(NY)
+            to = start_minute.tz_convert(NY)
+        elif size == 'day':
+            idx = all_sessions.get_loc(session_label)
+            start_session = all_sessions[idx - limit + 1]
+            _from = start_session.tz_localize(None).tz_localize('America/New_York')
+            to = session_label.tz_localize(None).tz_localize('America/New_York')
+
+        return _from, to
+
+    # move fetch function to here, cause the parallelize_with_multi_process don't support Closure
+    def _fetch(self, args):
+        @skip_http_error((404, 504))
+        def wrapper():
+            symbols = args[0]  # symbols can be list or str
+            _from = args[1]
+            to = args[2]
+            size = args[3]
+            if self._use_polygon:
+                assert isinstance(symbols, str)
                 df = self._api.polygon.historic_agg_v2(
-                    symbol, 1, size,
+                    symbols, 1, size,
```

Comment:
you're sending one symbol at a time. (check how you've built the list.)

Reply:
ok